homeassistant/components/neato/sensor.py

```diff
@@ -0,0 +1,102 @@
+"""Support for Neato sensors."""
+import logging
+
+from datetime import timedelta
+from pybotvac.exceptions import NeatoRobotException
+
+from homeassistant.components.sensor import DEVICE_CLASS_BATTERY
+from homeassistant.helpers.entity import Entity
+
+from .const import NEATO_ROBOTS, NEATO_LOGIN, NEATO_DOMAIN, SCAN_INTERVAL_MINUTES
+
+_LOGGER = logging.getLogger(__name__)
+
+SCAN_INTERVAL = timedelta(minutes=SCAN_INTERVAL_MINUTES)
+
+BATTERY = "Battery"
+
+
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+    """Set up the Neato sensor."""
+    pass
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Set up the Neato sensor using config entry."""
+    dev = []
+    for robot in hass.data[NEATO_ROBOTS]:
+        dev.append(NeatoSensor(hass, robot))
+
+    if not dev:
+        return
+
+    _LOGGER.debug("Adding robots for sensors %s", dev)
+    async_add_entities(dev, True)
+
+
+class NeatoSensor(Entity):
+    """Neato sensor."""
+
+    def __init__(self, hass, robot):
+        """Initialize Neato sensor."""
+        self.robot = robot
+        self.neato = hass.data.get(NEATO_LOGIN)
```

Comment:
Pass in the neato instance to the entity instead.

Reply:
I understand your intention. The other platforms are also initialized like this. Is that wrong too?

Reply:
Done, if we need to update the other platforms we should probably do that in a subsequent PR?